bundles/admin/admin-announcements/publisher/AnnouncementsTool.js

```diff
@@ -0,0 +1,357 @@
+
+            // Set the button handler
+            template.find('button').html(buttonLabel).on('click', function () {
+                if (me.isAnnouncementsDialogOpen === false) {
+                    me._openAnnouncementsDialog(jQuery(this));
+                }
+            });
+            if (me.annTitles.length > 0) {
+                template.find('input[name=publisher-announcements]').val(me.annTitles.toString());
+            }
+            return template;
+        },
+
+        /**
+         * Creates and opens the dialog from which to choose the announcements.
+         *
+         * @method _openAnnouncementsDialog
+         */
+        _openAnnouncementsDialog: function () {
+            const me = this;
+            const popup = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+            const closeButton = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            const title = me.localization.tool.popup.title;
+            const content = me.templates.announcementsPopup.clone();
+
+            closeButton.setTitle(me.localization.tool.popup.close);
+            closeButton.setHandler(function () {
+                popup.close(true);
+                me.isAnnouncementsDialogOpen = false;
+            });
+
+            me.announcements.forEach(announcement => {
+                const announcementInput = me.templates.inputCheckbox.clone();
+                const annName = announcement.title;
+                const annTime = announcement.begin_date.replace(/-/g, '/') + ' - ' + announcement.end_date.replace(/-/g, '/');
+
+                announcementInput.find('input[type=checkbox]').attr({
+                    'id': announcement.id,
+                    'name': 'announcement',
+                    'value': announcement.title
+                });
+                announcementInput.find('label').html(annName).attr({
+                    'for': announcement.id
+                });
+
+                if (me.isAnnouncementValid(announcement)) {
+                    content.find('div#ann-time').append('<div>' + annTime + '</div>');
+                    if (me.shouldPreselectAnnouncement(announcement)) {
+                        announcementInput.find('input[type=checkbox]').prop('checked', true);
+                    }
+                    content.find('div#ann-title').append(announcementInput);
+                } else {
+                    content.find('div#ann-time').append('<div>' + annTime + '<div class="icon-warning-light"></div></div>');
+                    announcementInput.find('input[type=checkbox]').prop('disabled', true);
+                    content.find('div#ann-title').append(announcementInput);
+                }
+            });
+
+            // WHAT TO DO WHEN ANNOUNCEMENTS ARE SELECTED
+            content.find('input[name=announcement]').on('change', function () {
+                var announcement = me.announcements.find(item => item.title === jQuery(this).val());
```

Comment:
This means that if you have two announcements with the same title like "Maintenance Break" both will be selected or removed. You have an id for pretty much every element in the DOM added by this functionality, why not refer to the id here?

Using id's is problematic and I usually request them to be removed in general. With a conflicting id somewhere else on the page (might be a totally different functionality) will break this this code/make it work unexpectedly. These problems are hard to debug once you hit them (usually years later and in a functionality that has nothing to do with announcements). Please consider refactoring the code removing id's from the DOM elements and finding a different way for referencing the elements (by searching them under a specific parent elements and using classes etc).